lib/phase/flatten.js

```diff
@@ -548,6 +548,40 @@ function getFullStageName({ stageName, type }) {
     return `${STAGE_PREFIX}${stageName}:${type}`;
 }
 
+/**
+ * Get stage teardown requires nodes
+ * @param  {Object} jobs            Jobs
+ * @param  {String} stageName       Stage name
+ * @return {Array}                  Teardown requires
+ */
+function getTeardownRequires(jobs, stageName) {
+    const stageJobsRequires = [];
+    const stageJobNames = [];
+
+    // Find jobs that belong to the stage
+    // Get requires for jobs that belong to the stage
+    Object.keys(jobs).forEach(jobName => {
+        if (jobs[jobName].stage && jobs[jobName].stage.name === stageName) {
+            if (Array.isArray(jobs[jobName].requires)) {
+                stageJobsRequires.concat(jobs[jobName].requires);
+            } else {
+                stageJobsRequires.push(jobs[jobName].requires);
+            }
+            stageJobNames.push(jobName);
+        }
+    });
+
+    // Get terminal nodes
+    const teardownRequires = stageJobNames.filter(
+        jobName =>
+            !stageJobsRequires.includes(jobName) &&
+            !stageJobsRequires.includes(`~${jobName}`) &&
+            !stageJobsRequires.includes(jobName.split('~')[0])
```

Comment:
`!stageJobsRequires.includes(jobName.split('~')[0]`
When would we have `~` after job name?

Reply:
it's if the jobName starts with `~` but the requires list does not have the prefix

Reply:
Do we allow `~` as a prefix for a job name?
![image](https://github.com/screwdriver-cd/config-parser/assets/2124590/d355b249-c2d9-47e4-ab2e-cdb660fddb4a)

Reply:
removed